arch/x86_64/mcount-dynamic.c

```diff
@@ -541,7 +541,7 @@ int mcount_patch_func(struct mcount_dynamic_info *mdi, struct sym *sym,
 	if (min_size < CALL_INSN_SIZE)
 		min_size = CALL_INSN_SIZE;
 
-	if (sym->size < min_size)
+	if (sym->size <= min_size)
```

Comment:
But this changes what users might want if it's bigger than the call size.  I think it's better to change the above condition like

```c
if (min_size < CALL_INSN_SIZE + 1)
    min_size = CALL_INSN_SIZE + 1;

if (sym->size < min_size)
    return result;
```

Reply:
Right. it's better to apply dynamic patching in such cases. I just updated the patch. In AArch64, it was already written as `<=` before so switched it back.